ui/src/app/shared/model/error.model.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * Generic App Error
+ */
+export class AppError {
+
+  /**
+   * Message
+   */
+  message: string;
+
+  /**
+   * Constructor
+   * @param {string} message
+   */
+  constructor(message: string) {
+    this.message = message;
+  }
+
+  /**
+   * ToString
+   *
+   * @returns {string}
+   */
+  toString() {
+    return this.message;
+  }
+
+}
+
+/**
+ * API Error
+ */
+export class AppErrorApi extends AppError {
```

Comment:
Misleading name IMO - Could we rename it to `HttpAppError` or `HttpAwareAppError?`